### PR TITLE
finish javascript

### PR DESCRIPTION
**Pull request by karlbaumhauer**

### CODE REVIEW

File: src/js/main.js [hunk 1]
```diff
@@ -1,6 +1,65 @@
...
+}
 
-animatedScrollTo(500);
\ No newline at end of file
+function mainReplyClick() {
+    const clickedButton = document.querySelectorAll('.button');
+    clickedButton.forEach(button => button.classList.toggle("button--clicked"));
+}
+
+function mainOpenContact() {
```
Comment (jurekbarth) [inline src/js/main.js]: This function might be better called `mainToggleContact` or even `toggleContact`

File: src/js/main.js [hunk 2]
```diff
@@ -1,6 +1,65 @@
...
+    const contact = document.querySelector('.mainConversion__contactBox');
+    contact.classList.toggle('mainConversion__contactBox--active');
+    const contactHeadline = document.querySelector('.mainConversion__contactBoxHeadline');
+    contactHeadline.classList.toggle('mainConversion__contactBoxHeadline--active');
+    const arrowUp = document.querySelector('.mainConversion__contactBoxArrowUp');
+    arrowUp.classList.toggle('mainConversion__contactBoxArrowUp--active');
```
Comment (jurekbarth) [inline src/js/main.js]: Do you really modify three dom elements? I would suggest to set a class on the block element, so you have only one dom modification.

File: src/js/main.js [hunk 3]
```diff
@@ -1,6 +1,65 @@
...
+    var box = document.querySelector('.mainConversion__contactBox');
+    if (box.classList.contains('mainConversion__contactBox--active')) {
+        document.getElementById('anchor1_1').scrollIntoView({behavior: "smooth"});
+    } else {
+        document.getElementById('anchor1_2').scrollIntoView({behavior: "smooth"});
+    }
+    mainOpenContact();
+}
+
+// Red Conversion
```
Comment (jurekbarth) [inline src/js/main.js]: Further on, this code seems repetitive to the other code here. In order to keep your code dry (don't repeat yourself) i would suggest to refactor this code to be a little bit more versatile to work with both classes/types of buttons, e.g. you could give your function `toggleContact` an argument `toggleContact(selector)` and then you only need one of these functions :) Does that make sense for you?

File: src/js/main.js [hunk 4]
```diff
@@ -1,6 +1,65 @@
...
+const redButtons = document.querySelectorAll('.redConversion__button');
+redButtons.forEach(redConversion__button => redConversion__button.addEventListener('click', handleRedButton));
+
+function handleRedButton() {
+    redScroll();
+    redReplyClick();
+}
+
+function redReplyClick() {
```
Comment (jurekbarth) [inline src/js/main.js]: Whatever that function does, i don't know if it makes sense. Why do you toggle classes on every `.button` class when you clicked on a conversion button? The same happens on line 10.

File: src/index.html [hunk 5]
```diff
@@ -6,39 +6,50 @@
...
         </div>
+        <div class="mainConversion__contactBoxArrowUp redConversion__contactBoxArrowUp"></div>
+        <div id="redConversion__contactBox" class="mainConversion__contactBox redConversion__contactBox">
+            <p class="redConversion__contactBoxHeadline redConversion__contactBoxHeadline">Contact us</p>
+        </div>
     </div>
     <div class="secondConversion">
         <h1 class="headline">CtA - Secondary conversion</h1>
         <div class="mainConversion__content-wrapper mainConversion__content-wrapper--secondcon">
             <div class="mainConversion__right--secondcon">
```
Comment (jurekbarth) [inline src/index.html]: I know it's not part of this PR but anyway, i think you're missing a class here `mainConversion__right`

File: src/index.html [hunk 6]
```diff
@@ -6,39 +6,50 @@
...
                 <p class="mainConversion__secondaryHeadline">A wonderful serenity has taken possessio</p>
                 <p>A wonderful serenity has taken possession of my entire soul, like these sweet mo</p>
-                <button class="button">
+                <button id="anchor1_2" class="button mainConversion__button">
                     Link button
                 </button>
             </div>
             <img src="img/IMG_0346.jpg" alt="Bergfoto" class="mainConversion__right">
         </div>
+        <div class="mainConversion__contactBoxArrowUp"></div>
```
Comment (jurekbarth) [inline src/index.html]: That's not really semantic html, you can either change that to a `button` element or at least give it a role like to `role=button`

File: src/scss/style.scss [hunk 7]
```diff
@@ -69,3 +69,77 @@ img {
     background-color: $secondary-fontcolor;
   }
 }
+
+.mainConversion__contactBox {
```
Comment (jurekbarth) [inline src/scss/style.scss]: You could make this even easier
```
.mainConversion {
  &__contactBox { ... }
  &__contactBoxHeadline { ... }
}
```

File: src/scss/style.scss [hunk 8]
```diff
@@ -69,3 +69,77 @@ img {
...
+  height: 0;
+  padding-left: 10%;
+  &--active {
+    padding-top: 1%;
+    height: 300px;
+    background-color: darkblue;
+  }
+}
+
+.redConversion__contactBox {
```
Comment (jurekbarth) [inline src/scss/style.scss]: Same as above, you could use SCSS nesting and concatenation with `&`

File: src/scss/style.scss [hunk 9]
```diff
@@ -69,3 +69,77 @@ img {
...
+
+.redConversion__contactBoxArrowUp {
+  width: 0;
+  height: 0;
+  border-left: 20px solid transparent;
+  border-right: 20px solid transparent;
+  border-bottom: 0px solid white;
+  top: -20%;
+  margin: 0 0 0 9%;
+  transition: all 0.6s ease; //weshalb klappt hier die transitiontime nicht???
```
Comment (jurekbarth) [inline src/scss/style.scss]: What do you want to accomplish and what is not working here? :)

File: src/js/main.js (hunk 2 above)
Comment (karlbaumhauer) [reply src/js/main.js]: I don't really get it. If I set a class on the block I still need to modify different style components an each of the three elements.

Comment (karlbaumhauer) [reply src/js/main.js]: I think I got it and changed it.

Comment (luke-m) [reply src/js/main.js]: DOM manupilation in this context is using JS to add and remove classes, which is "expensive" in terms of calculation. Having predefined styles which apply to certain selectors which are based on the block's modifier is "free".

File: src/js/main.js (hunk 4 above)
Comment (karlbaumhauer) [reply src/js/main.js]: realized it to late and fixed it...;-)

File: src/index.html (hunk 5 above)
Comment (karlbaumhauer) [reply src/index.html]: the class exists... its not in the div but in the image... 

it was in a div around the image but I had trouble to get rid of a bottom margin on the image and could not fix it until I removed the container from the image.

File: src/scss/style.scss (hunk 9 above)
Comment (karlbaumhauer) [reply src/scss/style.scss]: old comment and fixed since a few commits

**Pull request merged**